src/main/java/org/dsa/iot/alarm/Subscriptions.java

```diff
@@ -0,0 +1,199 @@
+/* THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
+ * TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
+ * NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
+ * DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
+ * IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+package org.dsa.iot.alarm;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.dsa.iot.dslink.DSLink;
+import org.dsa.iot.dslink.link.Requester;
+import org.dsa.iot.dslink.node.value.SubscriptionValue;
+import org.dsa.iot.dslink.util.handler.Handler;
+
+/**
+ * Manages multiple subscribers to the same path.
+ *
+ * @author Aaron Hansen
+ */
+class Subscriptions {
```

Comment:
This looks a bit like `SubscriptionPool` in the historian. Did you check if you could have used it?